### DISCUSSION TITLE
How to use the StateChanged event?

### DISCUSSION THREAD

**Discussion by codesinging** [spatie/laravel-model-states#215] [Ideas]
How to customize the state changed event?

Suggest to add configuration to set custom StateChanged event class.

**Comment by sirikkoster** [comment]
Please add an example

**Comment by fmeccanici** [comment]
I recently added this (see https://github.com/spatie/laravel-model-states/pull/237). Now you can do the following to register your custom event:

```php 
abstract class PaymentState extends State
{
    abstract public function color(): string;
    
    public static function config(): StateConfig
    {
        return parent::config()
            ->stateChangedEvent(CustomStateChanged::class)
        ;
    }
}
